### PR TITLE
LSP Code Action: Add missing patterns to inexhaustive case expression

### LINKED ISSUE #2957: LSP code action: add missing patterns to case expression
The exhaustiveness checker has this information so we need to expose it to the language server

It may be challenging to work out in the exhaustiveness checker if the variant names need to be qualified or not. It would be annoying to have to fix it every time.

Perhaps this needs to come after https://github.com/gleam-lang/gleam/issues/2952

### PR DESCRIPTION
**Pull request by GearsDatapacks**
Closes #2957 
I have added a code action which fills in the missing patterns for an inexhaustive case expression. The formatting of the resulting syntax isn't great at the moment, but it works.
Let me know if I should work on adapting the generated code actions to produce nicer looking code as per the situation.

NOTE: This currently doesn't work properly with multi-patterns due to #2426. But I think it's useful enough to have now, and it will work fine as soon as that issue is fixed.

### CODE REVIEW

File: CHANGELOG.md
Comment (lpil) [inline CHANGELOG.md:112]: Could you use a different example please 🙏  We don't want to recommend using tuples like this

File: compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_bool.snap
Comment (lpil) [inline compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_bool.snap:19]: This isn't very helpful. I would be very frustrated if the action gave me this.

Is there a way to add the constructors here?

File: compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_custom_type.snap
Comment (lpil) [inline compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_custom_type.snap:34]: Remove this blank line please 🙏

**Review by lpil — COMMENTED** [review]
Thank you! Some notes inline

File: compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_bool.snap
Comment (GearsDatapacks) [reply compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_bool.snap:19]: Unfortunately this is more a problem with the exhaustiveness checker. For this code:
```gleam
pub fn main() {
  case True {
    
  }
}
```

The compiler produces this error:
```
error: Inexhaustive patterns
  ┌─ /home/gears/projects/temp/testing_gleam/src/testing_gleam.gleam:5:3
  │  
5 │ ╭   case True {
6 │ │     
7 │ │   }
  │ ╰───^

This case expression does not have a pattern for all possible values. If it
is run on one of the values without a pattern then it will crash.

The missing patterns are:

    _
```

I was planning to open another issue after this about improving these suggestions, but should I open that now and leave this on hold until that is resolved?

Comment (lpil) [reply compiler-core/src/language_server/tests/snapshots/gleam_core__language_server__tests__action__add_missing_patterns_bool.snap:19]: Aye, I understand what the codebase does today. What I mean is that to be a useful addition this code action needs to be able to add constructors for empty case expressions.

**Comment by GearsDatapacks** [comment]
I've updated the parts which you suggested and are directly related to this PR. I guess I'll leave this as a draft for now, and open an issue to improve the exhaustiveness checker before continuing with this.

**Comment by lpil** [comment]
I've found a bug!

This code results in extra spacing
```rust
pub fn handle_request(request: Request) {
  case request.method {
    
  }
}
```
```rust
  case request.method {
    
  
    http.Connect -> todo
    http.Delete -> todo
    http.Get -> todo
    http.Head -> todo
    http.Options -> todo
    http.Other(_) -> todo
    http.Patch -> todo
    http.Post -> todo
    http.Put -> todo
    http.Trace -> todo
  }
```

**Comment by GearsDatapacks** [comment]
Alright, that case should be fixed now

**Comment by GearsDatapacks** [comment]
Actually there's one more case I want to fix before this is merged:
```gleam
let value = case
  some_very_long_variable_name_which_causes_the_formatter_to_wrap
 {}
```

Since it determines the indentation based on the start of the case expression, it thinks all the `let value = ` is indentation, and indents the patterns way too much

**Comment by GearsDatapacks** [comment]
That's all taken care of now

**Review by lpil — APPROVED** [review]
Thank you!!! I love this so much!

**Pull request merged**
